### PR TITLE
Mariadb docker

### PR DESCRIPTION
**Pull request by gmokki**

### CODE REVIEW

File: nflow-tests/src/test/java/io/nflow/tests/ArchiveTest.java [hunk 1]
```diff
@@ -36,6 +38,7 @@
   private static final int STEP_1_WORKFLOWS = 4;
   private static final int STEP_2_WORKFLOWS = 7;
   private static final int STEP_3_WORKFLOWS = 4;
+  private static final int ARCHIVE_TIMEOUT = 15;
```
Comment (efonsell) [inline nflow-tests/src/test/java/io/nflow/tests/ArchiveTest.java]: ofSeconds(15) here?

File: nflow-tests/src/test/java/io/nflow/tests/ArchiveTest.java [hunk 2]
```diff
@@ -47,13 +50,14 @@ public ArchiveTest() {
     super(server);
   }
 
-  @Test // (timeout = ARCHIVE_TIMEOUT)
+  @Test
   @Order(1)
   public void cleanupExistingArchivableStuff() {
-    archiveService.archiveWorkflows(DateTime.now(), 10);
+    assertTimeoutPreemptively(ofSeconds(ARCHIVE_TIMEOUT), () ->
+      archiveService.archiveWorkflows(DateTime.now(), 10));
```
Comment (efonsell) [inline nflow-tests/src/test/java/io/nflow/tests/ArchiveTest.java]: static import

File: nflow-tests/src/test/java/io/nflow/tests/AbstractNflowTest.java [hunk 3]
```diff
@@ -118,7 +120,16 @@ protected ListWorkflowInstanceResponse getWorkflowInstance(int id, String expect
     return wf;
   }
 
-
+  protected ListWorkflowInstanceResponse getWorkflowInstanceWithTimeout(int id, String expectedState, Duration timeout) throws InterruptedException {
```
Comment (efonsell) [inline nflow-tests/src/test/java/io/nflow/tests/AbstractNflowTest.java]: Duration timeout -> int timeoutSeconds? now there is ofSeconds(x) everywhere in tests

Comment (gmokki) [reply nflow-tests/src/test/java/io/nflow/tests/AbstractNflowTest.java]: The `Duration` can be of any unit. This code does now know the unit is was based on.

**Pull request merged**